### PR TITLE
✨ Feature ➾ implement `taq compile-all` for the LIGO and SmartPy plugin

### LINKED ISSUE #1576: 🚧 Dev ➾ Add `compile all`
### 🔗 Epic|Feature Link

_No response_

### 🚥 Definition of Done

- [ ] 🎯 Planning and Design
- [ ] 🏎️ Dev Implementation Complete
- [ ] ♾️ PR Merged
- [ ] 🏆 Product Acceptance

### 🧰 Work Description

Reimplement `compile all`

### ⚖️ Acceptance Criteria

✨ Functional: 

- [x] `taq compile-all` will compile all contracts with a `main` entrypoint for LIGO
- [x] `taq compile-all` will compile all contracts with at least one compilation target for SmartPy

### 🔭 Scope

🙅‍♂️ Out of Scope: 
- Do not clean up other stuff

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jchenche**
# 🌮 Taqueria PR

Fixes #1576

## 🪁 Description

Implemented `taq compile-all` for the LIGO and SmartPy plugin which compiles all LIGO main contracts (contracts with a main function) and all SmartPy contracts with at least one SmartPy compilation target, respectively.

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [ ] ⛱️ I have completed this PR template in full and updated the title appropriately
- [ ] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [ ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🦀 Automated tests have been written and added to this PR
- [ ] 🐬 I have commented my code, particularly in hard-to-understand areas
- [ ] 🤿 Corresponding changes have been made to all documentation
- [ ] 🐚 Required changes to the VScE have been made
- [ ] 🪸 Required updates to scaffolds have been made
- [ ] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

_Please include a summary of the changes to the codebase._ _Please also include relevant motivation and context for the change_
_(e.g. what was the existing behaviour, why did it break, etc.)_

## 🎢 Test Plan

_Please describe the testing strategy and plan for this PR. Keep this lightweight and anticipate any testing challenges_

## 🛸 Type of Change

- [ ] 🧽 Chore ➾
- [ ] 🛠️ Fix ➾
- [x] ✨ Feature ➾
- [ ] 👷 Refactor ➾
- [ ] 🧪 Pre-Release ➾
- [ ] 🚀 Release ➾

### CODE REVIEW

File: taqueria-plugin-ligo/_readme.eta [hunk 1]
```diff
@@ -60,6 +60,16 @@ Lastly, `taq compile hello.mligo` will compile `hello.mligo` and emit `hello.tz`
 
 The `--json` flag will make the task emit JSON-encoded Michelson instead of pure Michelson `.tz`
 
+## The `taq compile-all` Task
+
+Basic usage is:
+
+```shell
+taq compile-all
```
Comment (egarson) [inline taqueria-plugin-ligo/_readme.eta:68]: What is an "eta" file and why does it repeat the same information as the "md"?

File: taqueria-plugin-ligo/compile-all.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { compileContractWithStorageAndParameter, TableRow } from './compile';
+
+const isMainContract = (parsedArgs: Opts, contactFilename: string): Promise<boolean> =>
+	readFile(getInputFilename(parsedArgs, contactFilename), 'utf8')
+		.then(data => /(const|let|function)\s+main/.test(data));
+
+const compileAll = async (parsedArgs: Opts): Promise<void> => {
+	let p: Promise<TableRow[]>[] = [];
+
+	const contractFilenames = await glob(
```
Comment (egarson) [inline taqueria-plugin-ligo/compile-all.ts:14]: That's curious - an async `glob`

Comment (egarson) [inline taqueria-plugin-ligo/compile-all.ts:14]: That's curious - an async `glob`

File: taqueria-plugin-ligo/_readme.eta (hunk 1 above)
Comment (jchenche) [reply taqueria-plugin-ligo/_readme.eta:68]: Not sure, I am inheriting the design from others. It's basically the same as README but I was told that "eta" file is to be updated

**Review by egarson — APPROVED** [review]
LGTM

**Review by mweichert — APPROVED** [review]
LGTM Jimmy. I did have three concerns which we touched base on in Slack about, and I don't think that should hold back the PR:
1) The glob calls won't work if the end-user calls the `compile-all` task from a subdirectory of the taqueria project. Based on what we discussed, you'll handle this in a separate PR
2) There's some performance concerns I have over iterating over the files and awaiting the isMainContract call. I posted a possible solution in Slack, but again, this can probably be deferred.
3) There's unvalidated casts from parsedArgs to CompileOpts.

**Pull request merged**